### PR TITLE
Fix response handling and request for POST message

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @danehansen @dwick

### CODE REVIEW

File: src/endpoints/v1.es6.js [hunk 1]
```diff
@@ -983,11 +983,17 @@ class APIv1Endpoint {
           }
 
           return basePost(uri, options, this.request, (body) => {
-            if (body) {
+            if (body.json.data) {
```
Comment (dwick) [inline src/endpoints/v1.es6.js]: this scares me a little, sure there are no edge cases?

File: src/endpoints/v1.es6.js [hunk 2]
```diff
@@ -983,11 +983,17 @@ class APIv1Endpoint {
...
               return new Message(message).toJSON();
+            } else if (body.json.captcha) {
+              return body.json;
+            } else if (body.json.errors.length){
+              throw body.json.errors;
             } else {
-              return null;
+              // for successful new messages in compose
+              // no data is returned
+              return 'success';
```
Comment (dwick) [inline src/endpoints/v1.es6.js]: returning `null` seems better than a string that is never used.

**Comment by dwick** [comment]
:nail_care:

**Comment by curioussavage** [comment]
:haircut: 

added validators too

File: src/models/message.es6.js [hunk 3]
```diff
@@ -4,7 +4,23 @@ class Message extends Base {
...
+        return Base.validators.minLength(this.get('subject'), 1) &&
+               Base.validators.maxLength(this.get('subject'), 100);
+      }.bind(message),
+      to: function() {
+        return Base.validators.minLength(this.get('subject'), 1);
+      }.bind(message),
+    };
   }
-};
+}
```
Comment (dwick) [inline src/models/message.es6.js]: nit: `;`

**Comment by dwick** [comment]
nit, but :fish:.  we should also figure out how to have proper status codes at some point instead of 200ing all the things

**Pull request closed without merging**